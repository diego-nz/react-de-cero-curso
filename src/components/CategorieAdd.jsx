import React, { useState } from 'react';
import PropTypes from 'prop-types'

export const CategorieAdd = ( {setCategories} ) => {
    
    const [inputValue, setInputValue] = useState('');

    const handleInput = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue.trim().length > 0){
            // Accede a las categorias mediante callback para no pasarlas como props
            setCategories( cats => [inputValue, ...cats]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <h2>Add categorie</h2>
            <input type="text" value={inputValue} onChange={handleInput}/>
        </form>
    )
}

CategorieAdd.propTypes = {
    setCategories: PropTypes.func.isRequired
}
