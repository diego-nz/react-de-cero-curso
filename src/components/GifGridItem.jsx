import React from 'react';
import PropTypes from 'prop-types'

export const GifGridItem = ( { image, title } ) => {
    return (
        <figure className="card__item">
            <img src={image} alt={title} />
            <figcaption>{title}</figcaption>
        </figure>
    )
}
// import proptypes imrpcp
// Props requeridas
GifGridItem.propTypes = {
    image: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
}
