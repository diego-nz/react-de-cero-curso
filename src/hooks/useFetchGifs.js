import { useState, useEffect } from "react"
import { getGifs } from "../helpers/getGifs"

export const useFetchGifs = ( categorie ) => {

    const [state, setstate] = useState({
        data: [],
        loading: true
    })

    useEffect(() => {
        getGifs( categorie ).then( data => setstate( {data: data, loading: false} ));
    }, [categorie])

    return state

}