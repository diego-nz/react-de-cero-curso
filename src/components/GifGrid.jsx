import React from 'react'
// import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types'


export const GifGrid = ( { categorie } ) => {
    
    // const [images, setImages] = useState([]);

    // Con : se esta renombrando el key en la destructuracion
    const { data: images, loading } = useFetchGifs( categorie );
    // useEffect(() => {
    //     getGifs( categorie ).then( data => setImages( data ));
    // }, [categorie])


    return (
        <>
            <h3> {categorie} </h3>
            { loading && <h4 className="animate__animated animate__flash">Cargando...</h4> }
            <div className="card-container animate__animated animate__zoomInUp">
                {images.map((img) => 
                    <GifGridItem key={img.id} {...img}/>
                )}
            </div>
        </>
    )
}

GifGrid.propTypes = {
    categorie: PropTypes.string.isRequired
}
