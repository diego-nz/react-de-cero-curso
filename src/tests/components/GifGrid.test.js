import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas del componente gifgrid', () => {
    const categorie = 'naruto';
    
    it('debe mostrar el componente <GifGrid>', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow( <GifGrid categorie={categorie} /> );
        expect( wrapper ).toMatchSnapshot();
    });

    it('debe de mostrar items cuando se cargan imagenes de useFetchGifs', () => {
        const gifs = [
            {id: 1, title: 'titulo-1', image:'https://ruta/de/imagen/1.gif'},
            {id: 2, title: 'titulo-2', image:'https://ruta/de/imagen/2.gif'},
            {id: 3, title: 'titulo-3', image:'https://ruta/de/imagen/3.gif'},
        ]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow( <GifGrid categorie={categorie} /> );
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );

    });
});