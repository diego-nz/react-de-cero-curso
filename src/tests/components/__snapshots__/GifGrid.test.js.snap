// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente gifgrid debe de mostrar items cuando se cargan imagenes de useFetchGifs 1`] = `
<Fragment>
  <h3>
     
    naruto
     
  </h3>
  <div
    className="card-container animate__animated animate__zoomInUp"
  >
    <GifGridItem
      id={1}
      image="https://ruta/de/imagen/1.gif"
      key="1"
      title="titulo-1"
    />
    <GifGridItem
      id={2}
      image="https://ruta/de/imagen/2.gif"
      key="2"
      title="titulo-2"
    />
    <GifGridItem
      id={3}
      image="https://ruta/de/imagen/3.gif"
      key="3"
      title="titulo-3"
    />
  </div>
</Fragment>
`;

exports[`Pruebas del componente gifgrid debe mostrar el componente <GifGrid> 1`] = `
<Fragment>
  <h3>
     
    naruto
     
  </h3>
  <h4
    className="animate__animated animate__flash"
  >
    Cargando...
  </h4>
  <div
    className="card-container animate__animated animate__zoomInUp"
  />
</Fragment>
`;
